lib/ClassMover/Domain/Name/QualifiedName.php

```diff
@@ -7,7 +7,7 @@
 class QualifiedName
 {
     /**
-     * @param string[] $parts
+     * @param non-empty-array<string> $parts
```

Comment:
Is this correct or can the list of parts also be empty. Then functions like end and begin sometimes return false which we need to handle.

Reply:
i think this is fine

Reply:
but would be good to handle the error scenarios, ideally the class should be closed and it should not be possible for those error conditions to happen, but in practice it probably is.

Reply:
Yeah, as far as I know there is only the factory methods that make this object.